### PR TITLE
migrator: better error logging

### LINKED ISSUE #1365: migrator: better error handling
Currently we only check for validation errors when migrating chunnks

We need a consistent way to account for:

* Errors when transforming XML strings into dicts (in this case we do not have a recid)
* Exceptions occuring during `dojson` conversion (in this case we __do__ have a recid and we can use the same format as we use for ValidationErrors)

### PR DESCRIPTION
**Pull request by mihaibivol**
* Refactors migration logic into a single function
* Wraps every step that might fail due to corrupt data in try: except:
  blocks
* Adds logging messages that are aggregated by Sentry

Signed-off-by: Mihai Bivol <mihai.bivol@cern.ch>

Closes #1365

### CODE REVIEW

**Comment by mihaibivol** [comment]
For deployment we also need to set `LOGGING_SENTRY_CELERY = True`

File: inspirehep/modules/migrator/tasks.py [hunk 1]
```diff
@@ -204,31 +203,13 @@ def create_index_op(record):
...
-                    db.session.merge(prod_record)
-                    continue
-
-                index_queue.append(create_index_op(record))
-
-                prod_record.valid = True
-                db.session.merge(prod_record)
+           with db.session.begin_nested():
+                record = migrate_and_insert_record(raw_record)
+                if record:
```
Comment (jacquerie) [inline inspirehep/modules/migrator/tasks.py]: This would be a nice place to add optional index population.

**Comment by mihaibivol** [comment]
@jacquerie this is ready for review now and can be integrated :)

File: inspirehep/modules/migrator/tasks.py [hunk 2]
```diff
@@ -176,7 +177,6 @@ def continuous_migration():
                         errors = "ValidationError: Record {0}: {1}".format(
                             recid, e
                         )
-                        current_app.logger.warning(errors)
```
Comment (jacquerie) [inline inspirehep/modules/migrator/tasks.py]: Why are you already removing this? Isn't this part of #1371?

File: inspirehep/celery.py [hunk 3]
```diff
@@ -9,3 +9,6 @@
 from .factory import create_app
 
 celery = create_celery_app(create_app())
+# Trigger an app log message upon import. Somehow this makes Sentry logging
+# work with `get_task_logger(__name__)`.
+celery.flask_app.logger.info('Created Celery app')
```
Comment (jacquerie) [inline inspirehep/celery.py]: Since you are here... what about a license and `__future__` imports in this file?

**Comment by jacquerie** [comment]
LGTM, but needs a rebase and a minor touch.

File: inspirehep/modules/migrator/tasks.py (hunk 2 above)
Comment (mihaibivol) [inline inspirehep/modules/migrator/tasks.py]: I didn't know how to test the continous migration. This should be replaced with ` migrate_and_insert_record(raw_record)` anyway

**Pull request merged**